### PR TITLE
AM2R: fix instances of certain resources not being used together

### PR DESCRIPTION
**Pull request by Miepee**
Whoops, turns out we had quite a bit of these!

### CODE REVIEW

File: randovania/games/am2r/logic_database/The Tower.txt [hunk 1]
```diff
@@ -114,7 +114,7 @@ Extra - liquid_info: {'liquid_type': 0, 'liquid_level': 0}
   > Door to Plasma Beam Chamber Access
       Any of the following:
           Space Jump Wall
-          Hi-Jump Boots and Mid-Air Morph (Intermediate)
+          Hi-Jump Boots and Morph Ball and Mid-Air Morph (Intermediate)
```
Comment (Miepee) [inline randovania/games/am2r/logic_database/The Tower.txt:117]: I'm unsure right now if this is supposed to be a morph glide instead.

File: randovania/games/am2r/logic_database/The Tower.txt [hunk 2]
```diff
@@ -138,7 +138,8 @@ Extra - liquid_info: {'liquid_type': 0, 'liquid_level': 0}
           All of the following:
               Hi-Jump Boots
               Any of the following:
-                  Power Grip or Mid-Air Morph (Beginner) or Movement (Intermediate)
+                  Power Grip or Movement (Intermediate)
+                  Morph Ball and Mid-Air Morph (Beginner)
```
Comment (Miepee) [inline randovania/games/am2r/logic_database/The Tower.txt:142]: ditto

File: randovania/games/am2r/logic_database/The Tower.txt (hunk 1 above)
Comment (JeffGainsNGames) [reply randovania/games/am2r/logic_database/The Tower.txt:117]: glide
image

File: randovania/games/am2r/logic_database/The Tower.txt (hunk 2 above)
Comment (JeffGainsNGames) [reply randovania/games/am2r/logic_database/The Tower.txt:142]: also ditto
image

**Review by JeffGainsNGames — CHANGES_REQUESTED** [review]

**Review by JeffGainsNGames — APPROVED** [review]

**Pull request merged**